cmd/dockerd/daemon_unix.go

```diff
@@ -107,3 +110,25 @@ func wrapListeners(proto string, ls []net.Listener) []net.Listener {
 	}
 	return ls
 }
+
+func newBuildKit(config *config.Config, d *daemon.Daemon, sm *session.Manager) (*buildkit.Builder, error) {
```

Comment:
Do not duplicate `newBuildKit` that has the same content but separate the `cgroupParent` detection to its own func.